thorlcr/activities/hashdistrib/thhashdistribslave.cpp

```diff
@@ -2706,17 +2709,27 @@ class CSpill : implements IRowWriter, public CSimpleInterface
     {
         if (NULL == writer)
             return;
-        flush();
+        writer->flush();
+        spillFileIO->flush();
```

Comment:
However, I don't think CSpill::close() should be leaving spillFileIO active/open.
It will mean when spilling and bucket handler phases are involved, there will be lots of buckets that are finished (close()'d), but there writing file handles are left open (to these dangling spill files) which are no longer being written to.
Then, when they are read, there will be 2 file handles open to each bucket spill file 1) the dangling write handles, and the reader file handles.

It would be better to actually close it here, and close out the stats as it's being closed.
e.g. by passing a stats object to close().